### PR TITLE
Domains and Redirects Doc

### LINKED ISSUE #2703: Desk Doc on Issue
@pantheon-systems/cse 
Priority (High):

## Issue Description:
in light of the recent platform changes with HTTPS, I believe some of the php snippets on this documentation page: https://pantheon.io/docs/redirects/ need to be updated to align with the php snippet on this documentation page: https://pantheon.io/docs/guides/launch/redirects

## Suggested Resolution 

Update the redirects/ page to match the latest and greatest updates within the launch guide.

### LINKED ISSUE #2706: Redirect Incoming Requests Doc Update (CNAME Flattening)
Re: [Redirect Incoming Requests - redirect from www to the bare domain](https://pantheon.io/docs/redirects/#redirect-from-www-to-the-bare-domain/)

Priority (Low‚ Medium‚ High): Low

## Issue Description: 
As per our doc:

> If you prefer to use the bare domain, use the following code block and run your DNS settings through a service that supports CNAME flattening. For details, see Domains and DNS.

>Note
If you are running the site on a Pro plan or above with an SSL certificate, use the snippet below without configuring a CNAME flattening service.

In our current setup (Global CDN +free managed HTTPS), CNAME flattening is not required for redirecting www to the bare domain. They only need to add the appropriate redirect in settings.php or wp-config.php

## Suggested Resolution 

Update doc and add a note that CNAME flattening not required for sites on Global CDN.

### PR DESCRIPTION
**Pull request by rachelwhitton**
Closes #2706 
Closes #2703 

## Effect
PR includes the following changes:
- Describe various domain types in a single reference doc (platform, custom, vanity, primary)
- Puts focus on a single redirect snippet for HTTPS + primary domain (managed in a single source file)
  - Collapse additional redirect scenarios for less distractions
- Delete /docs/redirects
- Delete /docs/platform-domains
- Restore /docs/domains

## Remaining Work
- [x] Disable existing redirect from /docs/domains => /docs/guides/launch/domains
- [ ] /docs/redirects => /docs/domains
- [ ] /docs/platform-domains => /docs/domains

### CODE REVIEW

**Comment by rachelwhitton** [comment]
ping @ari-gold and @lauren-kelly thoughts?

File: source/_docs/domains.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
+---
+title: Domains and Redirects
+description: Understand platform and custom domains and learn how to redirect requests to an alternate Drupal or WordPress domain via PHP.
+tags: [redirects, variables, dns]
+categories: []
+---
+## Platform domains
+Every site on Pantheon is given multiple environments (Dev, Test, and Live), each with their own unique platform domains which are bound to isolated resources, and are uniquely addressable on the internet via Pantheon's own DNS. [Multidev](/docs/multidev/) environments get the same treatment. Platform domains are configured as subdomains, and are generated by prefixing the environment to the site name:
```
Comment (ari-gold) [inline source/_docs/domains.md]: active voice. Great place to start the doc. What do you think about making this a little less technical?

File: source/_docs/domains.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
+---
+title: Domains and Redirects
+description: Understand platform and custom domains and learn how to redirect requests to an alternate Drupal or WordPress domain via PHP.
+tags: [redirects, variables, dns]
+categories: []
+---
```
Comment (ari-gold) [inline source/_docs/domains.md]: Maybe a short introduction about what domains are

File: source/_docs/domains.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+### robots.txt
+Pantheon serves a default robots.txt that disallows crawlers on platform domains (`/*.pantheonsite.io`, `/*.pantheon.io`, `/*.gotpantheon.com`, and `/*.sites.my-agency.com`). Crawlers are allowed on the Live environment to requests served with a custom domain (e.g., `www.example.com`). If you attempt to access your live environment with a platform-issued domain, even if you have a domain associated with the environment, the default robots.txt will be served.
+
+Pantheon does not allow crawlers on Dev, Test, or Multidev environments. Adding a custom domain to an environment other than Live will not permit crawlers to that environment.
+
+## Custom domains
+Custom domains are domains registered at a third-party provider, for which you control DNS configurations. Connect a custom domain to the Site Dashboard and point DNS at Pantheon to trigger the [automated process of provisioning HTTPS](/docs/https/).
+
+A paid plan is required to connect custom domains to an environment in the Pantheon Site Dashboard, up to the following limits:
```
Comment (ari-gold) [inline source/_docs/domains.md]: Nice and clear way to communicate domain limits =)

File: source/_docs/domains.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    </tr>
+  </tbody>
+</table>
+
+<div class="alert alert-info" role="alert">
+<h4 class="info">Note</h4>
+<p markdown="1">Add all domains you want to resolve to Pantheon within the Site Dashboard for each respective environment, as described in [Launch Essentials](/docs/guides/launch/). Automatic resolution of domains and wildcards are not supported.</p></div>
+
+### Primary domain
+A primary domain refers to a single custom domain used to serve responses, regardless of originating URL of a request (e.g., `https://www.example.com`). Configure the primary domain as part of the going live process described in [Launch Essentials: Redirect to a Primary Domain](/docs/guides/launch/redirects/).
```
Comment (ari-gold) [inline source/_docs/domains.md]: Pantheon uses the term "primary domain" to refer to a domain to redirect all traffic. So for example, if you choose www.myawesomesite.com as your primary domain, then if a visitor types in myawesomesite.com into their browser (or any other domain you have connected to your site), they will end up www.myawesomesite.com. Configure the domain....

File: source/_docs/domains.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+We recommend using the www subdomain prefix for all sites. While it’s good for visitors and DNS to resolve both www and the domain itself, it's best practice to choose one or the other and redirect from www to non-www (or vice versa, your call). This optimizes SEO by avoiding duplicate content and prevents session strangeness, where a user can be logged in to one domain but logged out of other domains at the same time.
+
+## Vanity domains for organizations
+Pantheon Partners, Strategic Partners, Enterprise accounts, Resellers, and OEM Partners have the ability to replace `pantheonsite.io` for each environment on every site they run with a custom vanity domain.
```
Comment (ari-gold) [inline source/_docs/domains.md]: have the ability > can 
Technically it doesn't "replace" the platform domain. I think the language in /docs/vanity-domains might work

File: source/_docs/domains.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+### Command line conditionals
+All redirect logic run on Pantheon environments should include the `php_sapi_name() != "cli"` conditional statement to see if WordPress or Drupal is running via the command line. Otherwise, redirects kill the PHP process before Drush and WP-CLI is executed resulting in a silent failure:
+
+```bash
+[notice] Command: <site>.<env> -- 'drush <command>' [Exit: 1]
+[error]
+```
+
+### Redirect to HTTPS and the primary domain
```
Comment (ari-gold) [inline source/_docs/domains.md]: How about linking to the section in the go live guide so we don't have to maintain the same content in both places?

File: source/_docs/domains.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+Configure redirects to the primary domain with HTTPS in `settings.php` or `wp-config.php` if a visitor arrives either (1) on a different domain or (2) without HTTPS:
+
+{% include("redirects.twig")%}
+
+Attempting to visit any page with HTTP or a non-primary domain should redirect to a page with the primary domain and a “Secure” label.
```
Comment (ari-gold) [inline source/_docs/domains.md]: What is a "Secure" label?

File: source/_docs/domains.md (hunk 6 above)
Comment (rachelwhitton) [inline source/_docs/domains.md]: We're managing the contents of the snippets in a single source, and inserting it on both pages at build time. I think it's fine on both pages but defer to you. Let me know if you still think we should remove and link to the guide @ari-gold

**Pull request merged**